src/journal-remote/journal-gatewayd.c

```diff
@@ -885,6 +885,66 @@ static int request_handler_machine(
         return MHD_queue_response(connection, MHD_HTTP_OK, response);
 }
 
+static int request_handler_boots(
+                struct MHD_Connection *connection,
+                void *connection_cls) {
+
+        _cleanup_(MHD_destroy_responsep) struct MHD_Response *response = NULL;
+        RequestMeta *m = ASSERT_PTR(connection_cls);
+        int r;
+        _cleanup_free_ LogId *boots = NULL;
+        size_t n_boots;
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *array = NULL;
+        _cleanup_free_ char *json = NULL;
+
+        assert(connection);
+
+        r = open_journal(m);
+        if (r < 0)
+                return mhd_respondf(connection, r, MHD_HTTP_INTERNAL_SERVER_ERROR, "Failed to open journal: %m");
+
+        r = journal_get_boots(
+                        m->journal,
+                        false,
+                        SIZE_MAX,
```

Comment:
our current coding style suggests that "opaque" parameters should be prefixed with a comment saying what they are. i.e. these two should be prefixed like this:

```c
/* advance_older = */ false, 
/* max_ids = */ SIZE_MAX,
```

for the other params things are clear from the variable names you pass so no need there